olive/passes/onnx/genai_model_exporter.py

```diff
@@ -40,6 +40,30 @@ def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassCon
             )
         }
 
+    def validate_search_point(
+        self, search_point: Dict[str, Any], accelerator_spec: AcceleratorSpec, with_fixed_value: bool = False
+    ) -> bool:
+        if with_fixed_value:
+            search_point = self.config_at_search_point(search_point or {})
+        precision = search_point.get("precision")
+        device = (
+            Device.CPU
+            if self.accelerator_spec.execution_provider
+            in AcceleratorLookup.get_execution_providers_for_device(Device.CPU)
+            else Device.GPU
+        )
+        if precision == self.Precision.FP16 and device == Device.CPU:
+            logger.info(
+                "FP16 is not supported on CPU. Valid precision + execution"
+                "provider combinations are: FP32 CPU, FP32 CUDA, FP16 CUDA, INT4 CPU, INT4 CUDA"
+            )
+            return False
+        return True
+
+    @staticmethod
+    def is_accelerator_agnostic(accelerator_spec: AcceleratorSpec) -> bool:
+        return False
```

Comment:
Basically, hardware dependent means need create InferenceSession and need run on target.

Reply:
Yes, like Mike says, a search point can be invalid for a particular ep while the pass is accelerator agnostic.

Pass is only non-agnostic if the output of the pass changes based on the accelerator.

Reply:
This PR description #481 has more info on this.

Reply:
Actually, line 81 depends on the accelerator_spec so it satisfies the requirement.

Reply:
Seems, we need an extra pass config like `target_ep`. Then the `is_accelerator_agnostic` can be True.